src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,242 @@
+@Produces(MediaType.APPLICATION_JSON)
+public class CuEinvoiceApiResource {
+
+    private static final Log LOG = LogFactory.getLog(CuEinvoiceApiResource.class);
+
+    private LookupDao lookupDao;
+    private CuEinvoiceDao cuEinvoiceDao;
+    private Gson gson = new Gson();
+
+    @Context
+    protected HttpServletRequest servletRequest;
+
+    @Context
+    protected HttpServletResponse servletResponse;
+
+    @GET
+    public Response describeEinvoiceApiResource() {
+        return Response.ok(CUPurapConstants.Einvoice.EINVOICE_KFS_API_DESCRIPTION).build();
+    }
+
+    @GET
+    @Path("vendors")
+    public Response getVendors(@Context HttpHeaders headers) {
+        try {
+            List<String> vendorNumbers = getVendorNumbersFromRequest();
+            List<VendorDetail> vendors = getCuEinvoiceDao().getVendors(vendorNumbers);
+            List<Properties> vendorsSerialized = vendors.stream().map(vendor -> getVendorProperties(vendor)).collect(Collectors.toList());
+            return Response.ok(gson.toJson(vendorsSerialized)).build();
+        } catch (Exception ex) {
+            LOG.error("getVendors", ex);
+            return ex instanceof BadRequestException ? respondGetVendorsBadRequest() : respondInternalServerError(ex);
+        }
+    }
+
+    @GET
+    @Path("vendors/{vendorNumber}")
+    public Response getVendor(@PathParam(PurapPropertyConstants.VENDOR_NUMBER) String vendorNumber, @Context HttpHeaders headers) {
+        try {
+
+            HashMap<String, String> vendorCombinedPk = parseVendorNumber(vendorNumber);
+            VendorDetail vendorDetail = getLookupDao().findObjectByMap(VendorDetail.class.newInstance(), vendorCombinedPk);
+            if (ObjectUtils.isNull(vendorDetail)) {
+                return respondNotFound();
+            }
+            String responseBody = gson.toJson(getVendorProperties(vendorDetail));
+            return Response.ok(responseBody).build();
+        } catch (Exception ex) {
+            LOG.error("getVendor", ex);
+            return ex instanceof BadRequestException ? respondGetVendorBadRequest() : respondInternalServerError(ex);
+        }
+    }
+
+    @GET
+    @Path("po/{purapDocumentIdentifier}")
+    public Response getPurchaseOrder(@PathParam(PurapPropertyConstants.PURAP_DOC_ID) String purapDocumentIdentifier, @Context HttpHeaders headers) {
+        try {
+            HashMap<String, String> map = new HashMap<>();
+            map.put(PurapPropertyConstants.PURAP_DOC_ID, purapDocumentIdentifier);
+            PurchaseOrderDocument poDoc = getLookupDao().findObjectByMap(PurchaseOrderDocument.class.newInstance(), map);
+            if (ObjectUtils.isNull(poDoc)) {
+                return respondNotFound();
+            }
+            String responseBody = serializePoDocumentToJson(poDoc);
+            return Response.ok(responseBody).build();
+        } catch (Exception ex) {
+            LOG.error(ex);
+            return respondInternalServerError(ex);
```

Comment:
should this error handling also check for BadRequestException?

Reply:
No, the functions called from getPurchaseOrder will not produce a BadRequestException. getPurchaseOrder is simpler than getVendor and getVendors because there is no parsing of VendorNumber(s) in getPurchaseOrder.

Reply:
OK, thanks